src/utils/__tests__/getLintTargets.spec.ts

```diff
@@ -0,0 +1,151 @@
+import { getLintTargets } from '../getLintTargets';
+
+describe('getLintTargets', () => {
+  describe('when @key is given as field', () => {
+    it('given object, returns its keys', () => {
+      expect(getLintTargets({ a: null, b: true }, '@key')).toStrictEqual([
+        {
+          path: ['a'],
+          value: 'a',
+        },
+        {
+          path: ['b'],
+          value: 'b',
+        },
+      ]);
+    });
+
+    it('given array, returns its indicies', () => {
+      expect(getLintTargets(['foo', 'bar'], '@key')).toStrictEqual([
+        {
+          path: ['0'],
+          value: '0',
+        },
+        {
+          path: ['1'],
+          value: '1',
+        },
+      ]);
+    });
+
+    it('given primitive property, returns the whole input', () => {
+      expect(getLintTargets('abc', '0')).toStrictEqual([
+        {
+          path: [],
+          value: 'abc',
+        },
+      ]);
+    });
+  });
+
+  describe('when property path is given as field', () => {
+    it('given existing property, returns lint targets', () => {
+      expect(getLintTargets({ a: null }, 'a')).toStrictEqual([
+        {
+          path: ['a'],
+          value: null,
+        },
+      ]);
+
+      expect(getLintTargets({ foo: ['a'] }, 'foo[0]')).toStrictEqual([
+        {
+          path: ['foo', '0'],
+          value: 'a',
+        },
+      ]);
+
+      expect(getLintTargets(['foo'], '0')).toStrictEqual([
+        {
+          path: ['0'],
+          value: 'foo',
+        },
+      ]);
+
+      expect(getLintTargets({ a: void 0 }, 'a')).toStrictEqual([
+        {
+          path: ['a'],
+          value: void 0,
+        },
+      ]);
+    });
+
+    it('given non-existing property, returns the whole document', () => {
```

Comment:
@P0lip Either the test title is misleading or there's something I don't understand. Where is the "whole document" in the provided output?

Reply:
ah, I should have stated `input`